### PR TITLE
Add Rust reshim

### PR DESCRIPTION
**Pull request by ericholscher**
This is required for Cargo builds,
as I hit here:

https://app.readthedocs.org/projects/test-builds/builds/26564308/

### CODE REVIEW

File: readthedocs/doc_builder/director.py [hunk 1]
```diff
@@ -458,18 +460,28 @@ def run_build_commands(self):
             # Execute ``asdf reshim python`` if the user is installing a
             # package since the package may contain an executable
             # See https://github.com/readthedocs/readthedocs.org/pull/9150#discussion_r882849790
-            for reshim_command in reshim_commands:
+            for reshim_command in python_reshim_commands:
                 # Convert tuple/list into set to check reshim command is a
                 # subset of the command itself. This is to find ``pip install``
                 # but also ``pip -v install`` and ``python -m pip install``
-                if reshim_command.issubset(command.split()):
+                if python_reshim_command.issubset(command.split()):
```
Comment (humitos) [inline readthedocs/doc_builder/director.py:467]: The variable names are incorrect in these blocks. `python_reshim_command` is not defined.

File: readthedocs/doc_builder/director.py [hunk 2]
```diff
@@ -458,18 +460,28 @@ def run_build_commands(self):
...
                     environment.run(
                         *["asdf", "reshim", "python"],
                         escape_command=False,
                         cwd=cwd,
                         record=False,
                     )
 
+            # Do same for Rust
+            for reshim_command in rust_reshim_commands:
+                if rust_reshim_command.issubset(command.split()):
```
Comment (humitos) [inline readthedocs/doc_builder/director.py:477]: Same here

**Review by humitos — CHANGES_REQUESTED** [review]
The concept looks good, but I think the code is not working as expected. Tests and linting are failing as well --I suppose they are related.

File: readthedocs/doc_builder/director.py (hunk 1 above)
Comment (ericholscher) [reply readthedocs/doc_builder/director.py:467]: Heh yea, 🙃

**Review by humitos — APPROVED** [review]

**Pull request merged**